Source/ASNodeContext.mm

```diff
@@ -0,0 +1,61 @@
+//
+//  ASNodeContext.m
+//  AsyncDisplayKit
+//
+//  Created by Adlai Holler on 5/29/19.
+//  Copyright © 2019 Pinterest. All rights reserved.
+//
+
+#import <AsyncDisplayKit/ASNodeContext+Private.h>
+
+#import <AsyncDisplayKit/ASAssert.h>
+
+#if AS_TLS_AVAILABLE
+
+#import <stack>
+
+static thread_local std::stack<ASNodeContext *> gContexts;
+
+void ASNodeContextPush(unowned ASNodeContext *context) {
+  gContexts.push(context);
+}
+
+ASNodeContext *ASNodeContextGet() {
```

Comment:
Should we check for the experiment and return `nil` in this case if it's not enabled?

Reply:
Check my most recent push – I don't think so – since there's no way to push a context outside of the experiment anymore.